sql/analyzer/process.go

```diff
@@ -0,0 +1,166 @@
+package analyzer
+
+import (
+	"io"
+
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/plan"
+)
+
+func trackProcess(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, error) {
+	if !n.Resolved() {
+		return n, nil
+	}
+
+	if _, ok := n.(*queryProcess); ok {
+		return n, nil
+	}
+
+	var query string
+	if x, ok := ctx.Value(sql.QueryKey).(string); ok {
+		query = x
+	}
+
+	var typ = sql.QueryProcess
+	if _, ok := n.(*plan.CreateIndex); ok {
+		typ = sql.CreateIndexProcess
+	}
+
+	processList := a.Catalog.ProcessList
+	pid := processList.AddProcess(ctx, typ, query)
+
+	var seen = make(map[string]struct{})
+	wrapTable := func(t sql.Table) (sql.Table, error) {
+		name := t.Name()
+		if _, ok := seen[name]; ok {
+			return t, nil
+		}
+
+		var total int64 = -1
+		if counter, ok := t.(sql.PartitionCounter); ok {
+			count, err := counter.PartitionCount(ctx)
+			if err != nil {
+				return nil, err
+			}
+			total = count
+		}
+		processList.AddProgressItem(pid, name, total)
+
+		seen[name] = struct{}{}
+		return &processTable{t, func() {
+			processList.UpdateProgress(pid, name, 1)
+		}}, nil
+	}
+
+	n, err := n.TransformUp(func(n sql.Node) (sql.Node, error) {
+		switch n := n.(type) {
+		case *plan.ResolvedTable:
```

Comment:
If we don't keep this, we would wrap `ResolvedTable`s in `ResolvedTable`s in the next `case`

Reply:
then we can remove 
```
t, err := wrapTable(n.Table)
if err != nil {
	return nil, err
}
return plan.NewResolvedTable(t), nil
```
Because all the tables in resovled tables will be already wrapped, right?

Reply:
Nope, `sql.Table` is not a node and is not reached by `TransformUp`, only via `ResolvedTable`. The second case is for `sql.Table`s that are also nodes.